Add ability to customize git commit generation prompt and provider

hassoncs opened the pull request:
Now you customize the prompt and provider we use to generate git commit messages, just like the enhancePrompt!
This also fixes a bug where the supportPrompts couldn't be saved/discarded the same way as other settings. See 92fb8506bd7eddd09c00d3a592edc3ad2070a2d8

Fixes https://github.com/Kilo-Org/kilocode/issues/718

I didn't wrap the parts of the fix for https://github.com/Kilo-Org/kilocode/commit/92fb8506bd7eddd09c00d3a592edc3ad2070a2d8 in kilocode_change comments because I'm going to open a PR to fix it against Roo shortly!

Roo PR: https://github.com/RooCodeInc/Roo-Code/pull/4623

new option in the prompt settings
![2025-06-12 19 17 33](https://github.com/user-attachments/assets/c58f09c1-c9c0-4925-8d4c-79dcbf7231f2)

save/discard working now
![2025-06-12 19 17 56](https://github.com/user-attachments/assets/34ff06bb-a617-4882-8161-b76b8a68f5c2)

Commit:
Add support for commit message generation API configuration

Introduces a commit message API configuration ID to the global settings, IPC messages, and task commands. Utilize this configuration in the CommitMessageProvider to generate commit messages based on the specified config, with fallback to the default configuration if the specified one is not available or fails to load. This change provides flexibility for users to customize the API used for commit message generation.

Commit:
✨ feat(commit-message): add custom support for commit message generation

- Introduce a new support prompt for generating commit messages
- Enhance commit message customization in various languages
- Allow setting API configuration for commit message generation in settings

♻️ refactor(commit-message): optimize commit message provider logic

- Refactor prompt building logic for commit message generation
- Remove unnecessary initialization call in generateCommitMessage method

🌐 i18n(prompts): add translations for commit message generation

- Update language files to include commit message generation labels and descriptions

Commit:
refactor(settings): integrate support prompts into cached state system

Support prompts previously bypassed the standard settings workflow by
auto-saving immediately when changed. This refactoring integrates them
into the cached state system used by all other settings.

Changes:
- Add customSupportPrompts to cached state in SettingsView
- Update PromptsSettings to work with cached values and callbacks
- Fix reset functionality that was broken due to state merging
- Remove direct postMessage calls in favor of cached state updates
- Ensure save button highlights and unsaved changes can be discarded

Key fixes:
- PromptsSettings: Use delete instead of undefined for reset
- webviewMessageHandler: Allow empty objects for clearing all prompts
- ExtensionStateContext: Replace instead of merge customSupportPrompts
- Remove resetSupportPrompt message type (handled via cached state)

Benefits:
- Consistent behavior with other settings (save button workflow)
- Proper change detection and discard functionality
- Working reset that actually removes custom prompts
- Better UX with explicit save/discard actions instead of auto-save

Fixes: Support prompt reset functionality that was restoring previous values

Commit:
Pre-review cleanup

Commit:
Add changeset

chrarnoldus commented on webview-ui/src/components/settings/PromptsSettings.tsx:
```diff
@@ -184,6 +187,49 @@ const PromptsSettings = () => {
+								</div>
+							</div>
+						</div>
+					)}
```
Perhaps worth it to extract this to a separate file.

hassoncs commented on src/core/webview/webviewMessageHandler.ts:
```diff
@@ -855,24 +853,6 @@ export const webviewMessageHandler = async (provider: ClineProvider, message: We
 				vscode.window.showErrorMessage(t("common:errors.update_support_prompt"))
 			}
 			break
-		case "resetSupportPrompt":
```
Everything is now triggered thorugh the updateSupportPrompt. `updateSupportPrompt` sends all prompts so to reset a prompt, `updateSupportPrompt` just sends an empty key for that type

hassoncs commented on webview-ui/src/components/settings/SettingsView.tsx:
```diff
@@ -744,7 +762,7 @@ const SettingsView = forwardRef<SettingsViewRef, SettingsViewProps>(({ onDone, t
 
 					{/* About Section */}
-					{activeTab === "about" && <About telemetrySetting={"disabled"} setTelemetrySetting={() => {}} />}
+					{activeTab === "about" && <About telemetrySetting={"disabled"} setTelemetrySetting={() => { }} />}
```
You are right! My prettier wasn't quite right. I'll fix this, ty!

hassoncs commented on src/core/webview/webviewMessageHandler.ts:
```diff
@@ -840,13 +840,11 @@ export const webviewMessageHandler = async (provider: ClineProvider, message: We
 		case "updateSupportPrompt":
 			try {
-				if (Object.keys(message?.values ?? {}).length === 0) {
+				if (!message?.values) {
```
`updateSupportPrompt` always contains all prompts so if you want to "reset" all promps `message.values` will be an empty object. The old check prevents resetting all prompts because it no-ops if the values are empty, which they would be if you are resetting the last prompt type.

Commit:
Address PR Comments
* Update translation for description
* Run prettier on SettingsView

Commit:
✨ feat(settings): add commit message configuration component
  - create `CommitMessagePromptSettings.tsx` for commit message API config
  - integrate `CommitMessagePromptSettings` in `PromptsSettings`
  - refactor `PromptsSettings.tsx` to use new commit message component

♻️ refactor(settings): clean up extension state usage
  - adjust `SettingsView.tsx` to streamline extension state destructuring
  - remove redundant state handling for better readability

Commit:
Default the values check

hassoncs committed:
Fix the save/discard/revert flow for Prompt Settings (#4623)

Add support for the save/discard flow for support prompt setting page

Normally when you edit things on the settings pages, the save button lights up,
allowing you to discard your changes. Currently the prompts page doesn't support this flow-
the prompts are immediately saved when they change.

With this change, we use the normal cachedState system in the SettingView, allowing users to dicard changes to their prompts like any other setting.
This removed the need for the resetSupportPrompt event since we send the entire state of the support prompts (same as before).

Test plan:
* Manually verified prompts can be saved/discarded for different types of support prompts.
